src/bluecon/BlueConAPI.py

```diff
@@ -154,11 +154,37 @@ async def registerAppToken(self, active: bool) -> bool:
                                     headers = (await self.__getOrRefreshOAuthToken()).getBearerAuthHeader()) as response:
                 return response.status == 200
     
-    def startNotificationListener(self):
+    async def startNotificationListener(self, hass = None): #hass is an optional parameter for Home Assistant
         """Starts the notification listener to get notifications about calls"""
 
-        def listener_thread(blueConAPIClient: BlueConAPI):
+        def run_in_event_loop(coroutine):
+            if hass:
+                return asyncio.run_coroutine_threadsafe(coroutine, hass.loop).result()
 
+            loop = asyncio.get_running_loop()
+            if loop.is_running():
+                return asyncio.ensure_future(coroutine)
+
+            return asyncio.run_coroutine_threadsafe(coroutine, loop).result()
+
+        def on_notification(blueConAPIClient: BlueConAPI, notification: dict, data_message):
+            idstr = data_message.persistent_id
+            received_persistent_ids = []
+
+
+            received_persistent_ids = run_in_event_loop(blueConAPIClient.__notificationInfoStorage.retrievePersistentIds())
+
+            if received_persistent_ids is not None and any(idstr in x for x in received_persistent_ids):
+                return
+
+            # TODO I commented this because it fails for me in Home Assistant
+            #run_in_event_loop(blueConAPIClient.__notificationInfoStorage.storePersistentId(idstr))
```

Comment:
This needs to be fixed before merging to main, as without persisting the IDs that were received, they will not be sent with the login request and FCM will send them again to be processed.

Reply:
Yep, I agree. At least we have a good entry point to be testing where we receive notifications to Home Assistant